src/plugins/data_view_editor/public/data_view_editor_service.ts

```diff
@@ -17,30 +17,113 @@ import {
 } from '@kbn/data-views-plugin/public';
 
 import { RollupIndicesCapsResponse, MatchedIndicesSet, TimestampOption } from './types';
-import { getMatchedIndices, ensureMinimumTime, extractTimeFields } from './lib';
+import { getMatchedIndices, ensureMinimumTime, extractTimeFields, removeSpaces } from './lib';
 import { GetFieldsOptions } from './shared_imports';
 
+export const matchedIndiciesDefault = {
+  allIndices: [],
+  exactMatchedIndices: [],
+  partialMatchedIndices: [],
+  visibleIndices: [],
+};
+
+export interface DataViewEditorServiceConstructorArgs {
+  services: {
+    http: HttpSetup;
+    dataViews: DataViewsServicePublic;
+  };
+  requireTimestampField?: boolean;
+  initialValues: {
+    name?: string;
+    type?: INDEX_PATTERN_TYPE;
+    indexPattern?: string;
+  };
+}
+
 export class DataViewEditorService {
-  constructor(private http: HttpSetup, private dataViews: DataViewsServicePublic) {
+  constructor({
+    services: { http, dataViews },
+    initialValues: {
+      type: initialType = INDEX_PATTERN_TYPE.DEFAULT,
+      indexPattern: initialIndexPattern = '',
+      name: initialName = '',
+    },
+    requireTimestampField = false,
+  }: DataViewEditorServiceConstructorArgs) {
+    this.http = http;
+    this.dataViews = dataViews;
+    this.requireTimestampField = requireTimestampField;
+    this.type = initialType;
+    this.indexPattern = removeSpaces(initialIndexPattern);
+
+    // fire off a couple of requests that we know we'll need
     this.rollupCapsResponse = this.getRollupIndexCaps();
+    this.dataViewNames$ = from(this.loadDataViewNames(initialName));
+
+    // public observables
+    this.matchedIndices$ = this.matchedIndicesInternal$.asObservable();
+    this.rollupIndicesCaps$ = this.rollupIndicesCapsInternal$.asObservable();
+    this.isLoadingSources$ = this.isLoadingSourcesInternal$.asObservable();
+    this.loadingTimestampFields$ = this.loadingTimestampFieldsInternal$.asObservable();
+    this.timestampFieldOptions$ = this.timestampFieldOptionsInternal$.asObservable();
+    this.rollupIndex$ = this.rollupIndexInternal$.asObservable();
+
+    // when list of matched indices is updated always update timestamp fields
+    this.matchedIndices$.subscribe(() => this.loadTimestampFields());
```

Comment:
Is this service created more than once during Kibana's lifetime? If not, then I think we must unsubscribe somewhere. probably have a destroy method and make sure we call it in the same place where we created it.

Otherwise, this could be a memory leak (which also has side-effects )

Reply:
Ok, I can add a `destroy` method to the service